src/pushNotification.ts

```diff
@@ -0,0 +1,39 @@
+import walletApi from './wallet/api/walletApi';
+import { PushNotificationResult, PushRegisterRequestData, PushUnregisterRequestData, PushUpdateRequestData } from './wallet/types';
+import HathorWalletServiceWallet from './wallet/wallet';
+
+export class PushNotification {
+
+  /**
+   * Register the device to receive push notifications
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  static async registerDevice(wallet: HathorWalletServiceWallet, payload: PushRegisterRequestData): Promise<PushNotificationResult> {
```

Comment:
Why don't we add here `wallet.failIfWalletNotReady()`, like we had before?

Reply:
Because this is a private method. Should we turn it public? I don't see any problem about it. However the axiosInstance also checks if the wallet is ready, so we have a redundance here.